tests/config_test.go

```diff
@@ -68,6 +71,93 @@ var _ = Describe("[Serial][rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][le
 		}
 	}
 
+	withSecret := func(secretName string, volumeName string, customLabel ...string) libvmi.Option {
+		volumeLabel := ""
+		if len(customLabel) > 0 {
```

Comment:
nit: I understand this as an implementation of an optional parameter. This confuses me as a reader, because as a user of this function I could shove in more arguments which would get silently ignored.

Maybe convert this vararg type into a func that sets the label on the volumeSource. This way you could just add the function if required, but it would require a bit more boilerplate:

i.e.

```
withSecret := func(secretName string, volumeName string, volumeSourceOptions ...func(vs *v1.VolumeSource)) {
...
				vs = &v1.VolumeSource{
					Secret: &v1.SecretVolumeSource{
						SecretName:  secretName,
					},
				},
				for _, vsOption := range volumeSourceOptions {
					vsOption(&vs)
				} 
```

Usage would then be:

```
withSecret("mySecretName", "myVolumeName", func(vs *v1.VolumeSource){ vs.Secret.VolumeLabel = "myCustomLabel"})
```

OTOH you would get rid of the `if len(...` block here. And maybe a chance to move this to libvmi (or if there is a libvolume?)

Just a thought...